aiida/backends/tests/export_and_import.py

```diff
@@ -2191,3 +2191,198 @@ def test_export_import_of_log_entries(self):
+            self.reset_database()
+            import_data(export_file, silent=True)
+
+            # Get nodes and comments
+            qb = QueryBuilder()
+            qb.append(Node, tag='node', project=['uuid'])
+            qb.append(Comment, with_node='node', project=['uuid'])
+            nodes_and_comments = qb.all()
+
+            self.assertEqual(len(nodes_and_comments), len(self.comments))
+            for entry in nodes_and_comments:
+                self.assertEqual(len(entry), 2)  # 1 Node + 1 Comment
+
+                import_node_uuid = str(entry[0])
+                import_comment_uuid = str(entry[1])
+
+                self.assertIn(import_node_uuid, [calc_uuid, data_uuid])
+                if import_node_uuid == calc_uuid:
+                    # Calc node comments
+                    self.assertIn(import_comment_uuid, calc_comments_uuid)
+                else:
+                    # Data node comments
+                    self.assertIn(import_comment_uuid, data_comments_uuid)
+
+        finally:
+            shutil.rmtree(tmp_folder, ignore_errors=True)
+
+    def test_multiple_user_comments_for_single_node(self):
+        import os, shutil, tempfile
+        from aiida.orm import CalculationNode, Comment, User, Node, QueryBuilder
+        from aiida.orm.importexport import export
+
+        tmp_folder = tempfile.mkdtemp()
+
+        try:
+            # Create users, node, and comments
+            user_one = User.objects.get_default()
+            user_two = User(email="commenting@user.s").store()
+
+            node = CalculationNode().store()
+
+            comment_one = Comment(node, user_one, self.comments[0]).store()
+            comment_two = Comment(node, user_one, self.comments[1]).store()
+
+            comment_three = Comment(node, user_two, self.comments[2]).store()
+            comment_four = Comment(node, user_two, self.comments[3]).store()
+
+            # Get values prior to export
+            users_email = [u.email for u in [user_one, user_two]]
+            node_uuid = node.uuid
+            user_one_comments_uuid = [c.uuid for c in [comment_one, comment_two]]
+            user_two_comments_uuid = [c.uuid for c in [comment_three, comment_four]]
+
+            # Export nodes
+            export_file = os.path.join(tmp_folder, 'export.tar.gz')
+            export([node], outfile=export_file, silent=True)
+
+            # Clean database and reimport exported file
+            self.reset_database()
+            import_data(export_file, silent=True)
+
+            # Get node, users, and comments
+            qb = QueryBuilder()
+            qb.append(Node, tag='node', project=['uuid'])
+            qb.append(Comment, tag='comment', with_node='node', project=['uuid'])
+            qb.append(User, with_comment='comment', project=['email'])
+            entries = qb.all()
+
+            self.assertEqual(len(entries), len(self.comments))
+            for entry in entries:
```

Comment:
How do you ensure that you get back both users and all four comments?

Reply:
It was only inferred in this test, however I have updated the test to work with sets instead. Directly checking number of ex-/imported nodes, users, and comments pertaining to a specific user are equal before and after. Also that the uuids and emails are the same.

Reply:
I have also added some more comments to these tests.

Reply:
Thanks!